tm2/pkg/std/tx.go

```diff
@@ -60,6 +60,15 @@ func (tx Tx) ValidateBasic() error {
 	return nil
 }
 
+func (tx Tx) IsSponsorTx() bool {
```

Comment:
IMO, sponsortx is when:
- first msg is noop
- first msg signer != other ones

Reply:
https://github.com/gnolang/gno/pull/2209/commits/c0bef5ab23e746afb4a0dc6fcde2c29288337be5